src/fsharp/symbols/Symbols.fs

```diff
@@ -1832,57 +1839,65 @@ and FSharpMemberOrFunctionOrValue(cenv, d:FSharpMemberOrValData, item) =
         checkIsResolved()
         match d with 
         | P p -> 
-            [ [ for (ParamData(isParamArrayArg, isInArg, isOutArg, optArgInfo, _callerInfo, nmOpt, _reflArgInfo, pty)) in p.GetParamDatas(cenv.amap, range0) do 
-                // INCOMPLETENESS: Attribs is empty here, so we can't look at attributes for
-                // either .NET or F# parameters
-                let argInfo: ArgReprInfo = { Name=nmOpt; Attribs= [] }
-                yield FSharpParameter(cenv, pty, argInfo, None, x.DeclarationLocationOpt, isParamArrayArg, isInArg, isOutArg, optArgInfo.IsOptional) ] 
-               |> makeReadOnlyCollection  ]
-           |> makeReadOnlyCollection
-
-        | E _ ->  []  |> makeReadOnlyCollection
+            [ 
```

Comment:
Why not make it just use array here instead? Please correct me if I'm wrong, but if array creation is optimized in a similar way to how lists are created, it'd be quite straigforward here and would be somewhat better perf-wise.

We use `Count` properties on these parameter groups and using arrays gives it instantly, while it's linear for lists.

Speaking about these APIs in general, I don't see what does making returned collections readonly add here since in most cases new collections are created on every call and modifying the collections themselves won't change FCS state in any way.

Reply:
@auduchinok , casting to IReadOnlyCollection, allocates a single iterator at the time of iteration rather than creating an array and copying over all of the elements:
````
    let makeReadOnlyCollection (arr: seq<'T>) = 
        System.Collections.ObjectModel.ReadOnlyCollection<_>(Seq.toArray arr) :> IList<_>
````
This approach trades an iteration/allocation and copy of all elements against element access and size performance on the far end.  Without specific measurements it's impossible for us to confirm which approach is better.
However, my gut feel, is that @forki has it right, and I prefer the updated code to the original, but that is pretty unscientific :-),  I also doubt the change will be noticeable either way by a user.